client/config/config_test.go

```diff
@@ -27,7 +29,8 @@ func initClientContext(t *testing.T, envVar string) (client.Context, func()) {
 	home := t.TempDir()
 	clientCtx := client.Context{}.
 		WithHomeDir(home).
-		WithViper("")
+		WithViper("").
+		WithCodec(codec.NewProtoCodec(codectypes.NewInterfaceRegistry()))
```

Comment:
Yes given the interface registry we have our codec too.

Reply:
As far as relaxing goes, we could default to the gogoproto marshaler (if no codec is specified) but IDK how it would behave with Interface unpacking (let's assume WithInterfaceRegistry was not called). 

If it worked before with the default gRPC codec then I would assume it also would work this way.

Reply:
with orm landing I don't think we can default to gogoproto. I think the force is fine for a single release and but then most teams will have migrated anyways